### ISSUE TITLE
`code --remote` cannot open files without filename extensions

### ISSUE THREAD

**Issue by joliss** [microsoft/vscode-remote-release#5083]
Issue Type: <b>Bug</b>

The `code --remote` command can generally open files. But this doesn't work with files that don't have a filename extension (such as `README`). Steps to reproduce:

1. On the remote machine, create a sample project:

    ```
    remote $ cd ~/src
    remote $ mkdir test
    remote $ cd test
    remote $ git init
    remote $ echo This is README.md > README.md
    remote $ echo This is README > README
    remote $ ls -lA
    total 8
    drwxrwxr-x 7 ubuntu ubuntu 119 May 20 17:10 .git
    -rw-rw-r-- 1 ubuntu ubuntu  15 May 20 17:10 README
    -rw-rw-r-- 1 ubuntu ubuntu  18 May 20 17:10 README.md
    ```

2. On a fresh VS Code Insiders installation, install the `code-insiders` command, and install the remote extension.

3. On the local machine, open `README.md` using the `--remote` syntax. This works as expected.

    ```
    local $ code-insiders --remote ssh-remote+ojin /home/ubuntu/src/test/README.md
    ```

    (Assuming the remote host name is `ojin` and the username is `ubuntu`.)

    ![image](https://user-images.githubusercontent.com/524783/119021331-38424700-b997-11eb-9f0f-2e326abb35e9.png)

4. On the local machine, open `README` (without file name extension). It does connect to the remote (see bottom left corner), but doesn't open the file.

    ```
    local $ code-insiders --remote ssh-remote+ojin /home/ubuntu/src/test/README
    ```

    ![image](https://user-images.githubusercontent.com/524783/119021906-e4842d80-b997-11eb-98a8-8b7d81e0b5b7.png)

----------------

Extension version: 0.65.6
VS Code version: Code - Insiders 1.57.0-insider (29c61570a5b9a669f777bb28b5acd5c37d99edbe, 2021-05-20T05:41:52.156Z)
OS version: Darwin arm64 20.4.0
Remote OS version: Linux x64 5.11.0-16-generic
Remote OS version: Linux x64 5.11.0-16-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|16.00GB (2.02GB free)|
|Process Argv|--remote ssh-remote+ojin /home/ubuntu/src/test --crash-reporter-id 5bedd228-a602-4962-b587-56869604957d|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: ojin|
|OS|Linux x64 5.11.0-16-generic|
|CPUs|AMD Ryzen 7 5800X 8-Core Processor (16 x 2200)|
|Memory (System)|62.79GB (53.25GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: ojin|
|OS|Linux x64 5.11.0-16-generic|
|CPUs|AMD Ryzen 7 5800X 8-Core Processor (16 x 2200)|
|Memory (System)|62.79GB (53.25GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
pythonvspyt602:30291494
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30291487
pythontb:30258533
vspre833cf:30267465
pythonptprofiler:30281269
vshan820:30294714
vscorecescf:30290706
pythondataviewer:30285072
vscus158cf:30286554
pythonvsuse255:30303025
vscgsv2:30305091
vscorehov:30301224
3f5j7703:30294757
vscod805cf:30301675

```

</details>

**Comment by aeschli** [comment]
That's correct. because we need can't connect to the remote  before we open a window we need to do some guessing on whether the input path is a file or a folder.

There are some tricks:
- To force that it's a file, use the --goto mode
`code-insiders --remote ssh-remote+ojin --goto /home/ubuntu/src/test/README:0`
`:0` is the line number 0, but in fact you can also omit it and just have `--goto`

- To force it's a folder, end with a slash
`code-insiders --remote ssh-remote+ojin /home/ubuntu/folderwith.dot/`

**Comment by joliss** [comment]
Ah, that's why! The `--goto` trick works!

I also just found `--folder-uri` and `--file-uri` in the [Tips and Tricks](https://code.visualstudio.com/docs/remote/troubleshooting) section of the documentation:

    code --folder-uri "vscode-remote://ssh-remote+ojin/home/ubuntu/src/test"
    code --file-uri "vscode-remote://ssh-remote+ojin/home/ubuntu/src/test/README"

If this syntax can do everything the `--remote` syntax can do, maybe it would be the preferred alternative?

The only remaining reference to the `--remote` option is the docs is in the [WSL](https://code.visualstudio.com/docs/remote/wsl) section:

    code --remote wsl+<distro name> <path in WSL>

Perhaps we should be recommending `--folder-uri` and `--file-uri` here as well, if the `vscode-remote://` syntax works for WSL? I don't have Windows so I can't test it.

The "dot in the filename" heuristic seems like enough of a cause of surprising behavior that I'm almost wondering if it's worth deprecating `--remote` altogether in favor of the URI syntax?

**Comment by aeschli** [comment]
We added `--remote` as a more user friendly variant of `--folder-uri` and `--file-uri`, consistent to the way you open local files and folders.  We also rather not promote the  `vscode-remote` URI scheme. It's kind of an implementation details. 

`--remote` is also necessary for empty remote windows (`code--new-window --remote wsl+ubuntu`).

So, no plans to deprecated it. I documented 'the tricks' in the docs.
